admin/server/projects.go

```diff
@@ -2154,3 +2336,39 @@ func valOrDefault[T any](ptr *T, def T) T {
 	}
 	return def
 }
+
+func resourceNamesFromProto(resources []*adminv1.ResourceName) []database.ResourceName {
+	if len(resources) == 0 {
+		return nil
+	}
+
+	seen := make(map[string]struct{}, len(resources))
+	res := make([]database.ResourceName, 0, len(resources))
+	for _, r := range resources {
+		if r == nil || r.Type == "" || r.Name == "" {
+			continue
+		}
+		key := normalizeResourceKey(r.Type, r.Name)
+		if _, ok := seen[key]; ok {
```

Comment:
See earlier comments – I believe `database.ResourceName` can be used as a map key. Also, since it does dedupe in the `database` package, why does it also need it here? (could one place be enough?)

Reply:
correct not need here, just keeping in database pkg